src/actions.ml

```diff
@@ -463,41 +463,67 @@ let create_pipeline_summary ?summary_top pipeline_info pipeline_url =
   |> (match summary_top with Some text -> List.cons text | None -> Fn.id)
   |> String.concat ~sep:"\n\n"
 
-let run_ci_minimization ~bot_info ~comment_thread_id ~owner ~repo ~docker_image
-    ~target ~opam_switch ~failing_urls ~passing_urls ~base ~head =
-  git_run_ci_minimization ~bot_info ~comment_thread_id ~owner ~repo
-    ~docker_image ~target ~opam_switch ~failing_urls ~passing_urls ~base ~head
-  >>= function
-  | Ok () ->
-      GitHub_mutations.post_comment ~id:comment_thread_id
-        ~message:
-          (f
-             "Hey, I have detected that there was a failure in project %s (for \
-              commit %s) without any failure in the test-suite.\n\
-              I checked that the corresponding job for the base commit %s \
-              succeeded. Now, I'm trying to extract a minimal test case from \
-              this so that it can be added to the test-suite. I'll come back \
-              to you with the results once it's done."
-             target head base)
-        ~bot_info
-      >>= GitHub_mutations.report_on_posting_comment
-  | Error f ->
-      Lwt_io.printf "Error: %s\n" f
+type ci_minimization_info =
+  { target: string
+  ; full_target: string
+  ; docker_image: string
+  ; opam_switch: string
+  ; failing_urls: string
+  ; passing_urls: string }
+
+let run_ci_minimization ~bot_info ~comment_thread_id ~owner ~repo ~base ~head
+    ~ci_minimization_infos =
+  Lwt_list.map_s
+    (fun {target; opam_switch; failing_urls; passing_urls; docker_image} ->
+      git_run_ci_minimization ~bot_info ~comment_thread_id ~owner ~repo
+        ~docker_image ~target ~opam_switch ~failing_urls ~passing_urls ~base
+        ~head
+      >>= fun result -> Lwt.return (target, result))
+    ci_minimization_infos
+  >>= fun results ->
+  results
+  |> List.partition_map ~f:(function
+       | target, Ok () ->
+           Either.First target
+       | target, Error f ->
+           Either.Second (target, f))
+  |> Lwt.return
+
+type ci_minimization_job_suggestion_info =
+  { base_job_failed: bool
+  ; base_job_errored: string option
+  ; head_job_succeeded: bool
+  ; missing_error: bool
+  ; non_v_file: string option
+  ; job_kind: string
+  ; overlayed: bool }
 
-let minimize_failed_test ~bot_info ~owner ~repo ~pr_id ~base ~head
-    ~head_pipeline_summary ~base_pipeline_summary = function
-  | {name= full_name; summary= Some summary; text= Some text} ->
-      let extract_artifact_url job_name summary =
-        if string_match ~regexp:(f "\\[%s\\](\\([^)]+\\))" job_name) summary
-        then Some (Str.matched_group 1 summary ^ "/artifacts/download")
-        else None
+let ci_minimization_fetch_suggestion_info ~head_pipeline_summary
```

Comment:
It's not super clear what the goal of this function is from its name. In particular the `fetch` part is a bit surprising.

Reply:
How about `ci_minimization_extract_job_specific_info`?  In the way I've set things up here, first we fetch and process all the information about the PR and the jobs, and only after that do we select which jobs to minimize.  I've separated the function that fetches and processes information common to all jobs in the PR from information specific to individual jobs, and this function is about the latter, that is, processing the checks text to extract information about a particular job.

Reply:
Should I add a documentation comment?

Reply:
Yes, sounds good.

Reply:
Done